log/options.go

```diff
@@ -87,3 +90,10 @@ func TraceOption(val bool) Option {
 		cfg.StackTrace = val
 	}
 }
+
+// JSONMarshalOption add option to configure custom JSON encoding
+func JSONMarshalOption(f func(v interface{}) ([]byte, error)) Option {
```

Comment:
Any reason this does not take a `json.Marshaler`?

```suggestion
func JSONMarshalOption(f func(v any}) ([]byte, error)) Option {
```

Reply:
refer to https://github.com/rs/zerolog/blob/master/encoder_json.go#L21C18-L21C21 and https://github.com/go-resty/resty/blob/v2/client.go#L123 practices